### DISCUSSION TITLE
Url remapping

### DISCUSSION THREAD

**Discussion by giovanni-motterle-mia-care** [mia-platform/community#430] [Q&A] [answered]
### What product are you having troubles with?

Console

### What Console version are you using?

1.96.3

### Description

Hi,
I was wondering how to use maps-proxyUrl.before.map and maps-proxyName.before.map to rewrite a path that is not completely know.
E.g. I would like to remap a call to /${some-variale-tag}/my-endpoint to /my-endpoint and then route /my-endpoint to /my-endpoint of a service "my-service".

Any ideas?

Thanks,
Giovanni

**Comment by fredmaggiowski** [comment]
Hi, you can use nginx-compatible regexes to match variable path tokens 


For a regex example see [this doc page](https://docs.mia-platform.eu/docs/development_suite/api-console/advanced-section/api-gateway/how-to#how-to-forward-a-request-to-another-url) that provides a few examples, such as: 

```nginx
"~^GET-/files/images/avatar/(?<username>[\w]+)/$" "/user/$username/avatar/"
```


Out of curiosity: why aren't you using a Console Endpoint with the path rewrite definition in design?

**Reply by giovanni-motterle-mia-care** [reply]
I have a microservice that exposes only one endpoint (let's say GET /my-endpoint) and I want my FE to be able to call that endpoint with a variable prefix (like GET /0.1.0/my-endpoint but also /0.2.0/my-endpoint).
As far as I know, console endpoints doesn't allow you to put public variables in their path definition or regular expressions. Am I right?

**Reply by fredmaggiowski** [reply, marked as answer]
>As far as I know, console endpoints doesn't allow you to put public variables in their path definition or regular expressions. Am I right?

You are correct, you are not allowed to use public variables interpolation in endpoint path rewrite. You can use path parameters though, by defining something like this:

<img width="601" alt="immagine" src="https://github.com/mia-platform/community/assets/7142570/620c5a5c-1078-4beb-b975-d672a30a6784">

**Reply by giovanni-motterle-mia-care** [reply]
ok, thanks !
